modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/AnnotationProcessorUtils.java

```diff
@@ -84,24 +92,48 @@ private AnnotationProcessorUtils()
     }
 
     /**
-     * Scans all classes in the {@code roundEnvironment} looking
-     * for methods annotated with {@link Operation}.
+     * Scans all the classes annotated with {@link Extension}, takes
+     * the {@link Operations} from those and returns the methods
+     * which are public and are not annotated with {@link Ignore}
      *
      * @param roundEnvironment the current {@link RoundEnvironment}
      * @return a {@link Map} which keys are the method names and the values are the
      * method represented as a {@link ExecutableElement}
      */
-    static Map<String, ExecutableElement> getOperationMethods(RoundEnvironment roundEnvironment)
+    static Map<String, Element> getOperationMethods(RoundEnvironment roundEnvironment, ProcessingEnvironment processingEnvironment)
     {
-        ImmutableMap.Builder<String, ExecutableElement> methods = ImmutableMap.builder();
-        for (Element rootElement : roundEnvironment.getRootElements())
+        ImmutableMap.Builder<String, Element> methods = ImmutableMap.builder();
+        for (Element rootElement : roundEnvironment.getElementsAnnotatedWith(Extension.class))
         {
             if (!(rootElement instanceof TypeElement))
             {
                 continue;
             }
 
-            methods.putAll(getMethodsAnnotatedWith((TypeElement) rootElement, Operation.class));
+            Class<?> extension = classFor((TypeElement) rootElement, processingEnvironment);
+            Class<?>[] operationsClasses = extension.getAnnotation(Operations.class).value();
+
+            List<AnnotationValue> annotationValues = IntrospectionUtils.getAnnotationFieldValue(rootElement, Operations.class, VALUE);
+            for (Class<?> operationClass : operationsClasses)
+            {
+                Element operationClassElement = annotationValues.stream()
```

Comment:
make the code more readable by extracting this to a private method with a descriptive name like getElementForClass()